src/osd/PG.cc

```diff
@@ -6245,32 +6180,23 @@ void PG::_delete_some()
     // final flush here to ensure completions drop refs.  Of particular concern
     // are the SnapMapper ContainerContexts.
     {
-      ObjectStore::Transaction t;
       PGRef pgref(this);
-      t.register_on_commit(new ContainerContext<PGRef>(pgref));
-      t.register_on_applied(new ContainerContext<PGRef>(pgref));
-      osd->store->queue_transaction(ch, std::move(t));
+      PGLog::clear_info_log(info.pgid, t);
+      t->remove_collection(coll);
```

Comment:
You'll know the ObjectStore interfaces better than me, but we're losing some guarantees here (we don't check the return value on ```queue_transaction()```) and unifying the object unlinks and the ```remove_collection``` op into a single transaction is probably a lot more work to force through in one go. Any idea why the ops were split to begin with?

Reply:
queue_transaction should be void anyway (all/most callers assert on non-zero).  The remove_collection is cheap, so no problem batching.  They were separate before just because of the weird way the code evolved.  This cleanup is partly cleaning up the dispatch_context/discard_context mess from before.